cirq-core/cirq/experiments/benchmarking/parallel_xeb.py

```diff
@@ -0,0 +1,684 @@
+            xeb_circuits.append(
+                attrs.evolve(self, wide_circuit=xeb_circuit, cycle_depth=cycle_depth)
+            )
+        return xeb_circuits
+
+
+def _target_to_operation(target: _TARGET_T) -> cirq.Operation:
+    if isinstance(target, ops.Gate):
+        return target(*devices.LineQid.for_gate(target))
+    elif isinstance(target, circuits.AbstractCircuit):
+        return circuits.CircuitOperation(target.freeze())
+    return target
+
+
+def _canonize_target(
+    target: Union[_TARGET_T, Dict[_QUBIT_PAIR_T, _TARGET_T]],
+) -> _CANONICAL_TARGET_T:
+    if isinstance(target, (ops.Gate, ops.Operation, circuits.AbstractCircuit)):
+        return _target_to_operation(target)
+    return {k: _target_to_operation(v) for k, v in target.items()}
+
+
+def _transform_moment_with_circuit_ops_to_moment_with_single_op(
+    moment: circuits.Moment,
+) -> circuits.Moment:
+    """Merges all circuit operations in a moment into a single circuit operation.
+
+    Args:
+        moment: A cirq moment composed of single and two qubit operations.
+
+    Returns:
+        A Moment with at most one CircuitOperation.
+    """
+    circuit_ops = [
+        op.mapped_circuit() for op in moment if isinstance(op, circuits.CircuitOperation)
+    ]
+    not_circuit_ops = [op for op in moment if not isinstance(op, circuits.CircuitOperation)]
+    all_subcircuits = circuit_ops
+    if not_circuit_ops:
+        all_subcircuits.append(circuits.Circuit(circuits.Moment(not_circuit_ops)))
+    return circuits.Moment(circuits.CircuitOperation(circuits.FrozenCircuit.zip(*all_subcircuits)))
+
+
+def create_combination_circuits(
+    circuit_templates: Sequence[cirq.Circuit],
+    combinations_by_layer: Sequence[rqcg.CircuitLibraryCombination],
+    target: _CANONICAL_TARGET_T,
+) -> Sequence[XEBWideCircuitInfo]:
+    """Zips two-qubit circuits into a single wide circuit for each of the given combinations.
+
+    Args:
+        circuit_templates: A sequence of narrow circuits.
+        combinations_by_layer: A sequence of combinations.
+        target: The target 2Q operation.
+
+    Returns:
+        A sequence of XEBWideCircuitInfo representing the wide circuits.
+    """
+    wide_circuits_info = []
+    for layer_comb in combinations_by_layer:
+        pairs = layer_comb.pairs
+        if isinstance(target, dict):
+            pairs = [pair for pair in pairs if pair in target]
+            assert pairs
+        for comb in layer_comb.combinations:
+            wide_circuits_info.append(
+                XEBWideCircuitInfo.from_narrow_circuits(
+                    circuit_templates,
+                    permutation=comb,
+                    pairs=layer_comb.pairs,  # type: ignore[arg-type]
+                    target=target,
+                )
+            )
+    return wide_circuits_info
+
+
+def simulate_circuit(
+    simulator: cirq.Simulator,
+    circuit: cirq.Circuit,
+    cycle_depths: Sequence[int],
+    circuit_id: Optional[int] = None,
```

Comment:
Please delete the unused `circuit_id`.  To preserve result order at lines 341-355 below, you can store future indices as `task_index = {t: i for i, t in enumerate(tasks)}` and get them back in the futures.as_completed loop as `task_index[result]`.  Or just use `pool.map` which follows the inputs order.

Reply:
done